libraries/Microsoft.Bot.Connector/Authentication/JwtTokenValidation.cs

```diff
@@ -25,8 +26,8 @@ public static class JwtTokenValidation
         /// validation may require outbound calls for Endorsement validation and other checks. Those calls are made to
         /// TLS services, which are (latency wise) expensive resources. The httpClient passed in here, if shared by the layers
         /// above from call to call, enables connection reuse which is a significant performance and resource improvement.</param>
-        /// <returns>Nothing</returns>
-        public static async Task AssertValidActivity(Activity activity, string authHeader, ICredentialProvider credentials, HttpClient httpClient = null)
+        /// <returns>Claims identity. An anonymous identity is returned if Authentication is disabled.</returns>
+        public static async Task<ClaimsIdentity> EnsureValidActivity(IActivity activity, string authHeader, ICredentialProvider credentials, HttpClient httpClient = null)
```

Comment:
There is another method, ValidateAuthHeader that is more what you're looking for. That may be a better candidate for this change.

Reply:
Let me refactor this class and bring it in sync with requirement. Agreed this became a little bit weird :D.